lib/health_cards/health_card_constraints.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+require 'json'
+require 'pp'
+require 'hashie'
+require 'base64'
+require 'zlib'
+require 'json/minify'
+
+
+## Functionality for "Health Cards are Small section of the Smart Health Cards Specification"
+module HealthCardConstraints
+  
+  # Remove Extraneous Fields from FHIR Bundle
+  def strip_fhir_bundle(bundle)
+
+    entries = bundle['entry']
+
+    url_map = Hash.new
+    resource_count = 0;
+
+    # Bundle.entry.fullUrl should be populated with short resource-scheme URIs (e.g., {"fullUrl": "resource:0})
+    entries.each do |entry|
+      old_url = entry["fullUrl"]
+      new_url = "Resource:#{resource_count}"
+
+      url_map[old_url] = new_url
+      entry["fullUrl"] = new_url
+
+      resource_count += 1
+    end
+
+    entries.each do |entry|
+
+      resource = entry['resource']
+
+      # Update references to match new resource-scheme URIs
+      HealthCardConstraints.update_links(resource, url_map)
+
+      # Remove Resource.id elements
+      resource.delete("id")
+      # Remove Resource.meta elements
+      resource.delete("meta")
+      # Remove Resource.text elements
+      resource.delete("text")
+  
+      resource.each do |element, value|
```

Comment:
This one I was a little unsure about. According to the [Health Cards are Small](https://github.com/smart-on-fhir/health-cards/blob/main/docs/index.md#health-cards-are-small) section of the spec, I interpreted the `id`, `meta`, and `text` deletions as just from the top level of the resources, since they were written as `Resource.x`. Does this make sense, or do you think I need to look more deeply for all `text` fields?
	
For the `CodeableConcept.text` and `Coding.display`, I made the assumption that they would only appear in the top level, but thinking about it more now, it makes sense that I should assume they could be nested more deeply within a resource, since they are just data types. Should I use the same recursive method to identify `CodeableConcept` and `Coding` as I do for reference?

Reply:
> Should I use the same recursive method to identify CodeableConcept and Coding as I do for reference?

Yes. This comment was only referring to this part, not the `id`, `meta`, and `text` part.

In the future (not as part of this PR) it may be worth creating a method that recursively navigates through nested hashes and executes a particular block on each element.